src/solver/mha/mha_solver_forward.cpp

```diff
@@ -65,8 +58,11 @@ constexpr S RoundUpToMultiple(T val, T mul)
     return Ceil(val, mul) * mul;
 }
 
-constexpr uint32_t nextPow2(uint32_t v)
+template <typename T>
+constexpr T nextPow2(T v)
 {
+    static_assert(std::is_integral_v<T>);
```

Comment:
[R] I am afraid this works correctly only with unsigned types (or positive signed values). If I am right then we need `assert(v>=0)` for signed integer types.

Reply:
Changed